src/EventHandler/AbstractMedia.php

```diff
@@ -0,0 +1,47 @@
+<?php declare(strict_types=1);
+
+/**
+ * This file is part of MadelineProto.
+ * MadelineProto is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+ * MadelineProto is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with MadelineProto.
+ * If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author    Daniil Gentili <daniil@daniil.it>
+ * @copyright 2016-2023 Daniil Gentili <daniil@daniil.it>
+ * @license   https://opensource.org/licenses/AGPL-3.0 AGPLv3
+ * @link https://docs.madelineproto.xyz MadelineProto documentation
+ */
+
+namespace danog\MadelineProto\EventHandler;
+
+use ReflectionClass;
+use JsonSerializable;
+use ReflectionProperty;
+use danog\MadelineProto\MTProto;
+use danog\MadelineProto\Ipc\IpcCapable;
+
+/**
+ * Represents a class for non-downloadable media.
+ */
+abstract class AbstractMedia extends IpcCapable implements JsonSerializable
```

Comment:
Please remove this class.

Reply:
I want to add non downloadable medias.

Like contact, dice, poll

If they are not media then what are them?

Reply:
They're not media, they're polls, dices, et cetera: the API may classify them as media files, but I don't think they should be treated as media files by the Event handler API.

Reply:
So you mean we will have a lot property like
```public readonly $dice``` ?

Reply:
yep!

Reply:
Though Dices are actually downloadable media (animated stickers), so they should be treated as media

Reply:
And contacts too (as VCF files)

Reply:
> And contacts too (as VCF files)

we have a `IpcCapable` class.
if for example Contact class does not call `parent::__construct` we have problems.
if calls, then what about fileId & etc properties?

Reply:
It's ok to make Contact & Dice extend Media